data-model-layer/data-model.md

```diff
@@ -91,9 +91,29 @@ Boolean is a scalar kind.  Its cardinality is two -- either the value 'true' or
 
 #### String kind
 
-String is a scalar kind.  Its cardinality is infinite -- strings do not have a length limit.
+Strings do not have any guarantees or requirements with regard to encoding.
 
-Strings should be UTF-8 text, in [NFC](https://www.unicode.org/reports/tr15/#Norm_Forms) canonicalization.
+All the hazards you typically find among programming languages and libraries working with binary
+serialization of strings apply to strings in the IPLD data model. IPLD's data model is conceptual,
+it takes the world as it is, and the world of strings has widely known compatibility issues.
+
+* Some languages/libraries guarantee a string encoding (typically UTF8), some do not.
+* Some languages/libraries can handle arbitrary byte data in strings, some do not.
+
+While some codec specifications will define a required encoding it should be noted that in practice
+many codec implementations leave this kind of validation and sanitizaton up to the consumer (application
+code) and it is typical to find arbitrary byte data in strings even in codecs that explicitely forbid it.
```

Comment:
To me this language is too weak. I'd right (the emphasis shouldn't be part of this PR, it's just to make it easier to parse if you read my comment):

While *most* [I'd say all, but I can see that that's too strong] define a required encoding, it should be noted in practice that *some* codec implementations leave this kind of validation and sanitization up to the consumer (application
code) and you *might* find arbitrary byte data in strings even in codecs that explicitly forbid it.

If you want a detailed write up, let me know. The short version is, I've looked into the all the language independent codecs I could find and *all* of them specify how strings should be like (JSON, CBOR, MessagePack, ProtocolBuffers, Cap'n'proto, CUE, Apache Thrift, Apache Avro, Flatbuffers, BARE, BSON, Borsh, Amazon Ion).